UOP1_Project/Assets/Scripts/Cutscenes/DialogueTimeline/DialogueMixerBehaviour.cs

```diff
@@ -0,0 +1,132 @@
+using UnityEngine.Playables;
+using UnityEngine.Timeline;
+
+namespace UOP1.Cutscenes
+{
+	public class DialogueMixerBehaviour : PlayableBehaviour
+	{
+		private const float k_StopThreshold = 0.1f;
+
+		private DialogueBinder m_dialogueBinder;
+		private bool m_initialized;
+		private int m_currentClipIndex;
+
+		private enum ClipState
+		{
+			Pending, Started, Ended
+		}
+
+		private class ClipData
+		{
+			public ScriptPlayable<DialogueBehaviour> Input;
+			public ClipState state;
+			public Range Range;
+		}
+		private ClipData[] m_clipsData;
+		private TimelineClip[] m_clips;
+
+		private struct Range
+		{
+			public double Start;
+			public double End;
+		}
+
+		public override void OnBehaviourPlay(Playable playable, FrameData info)
+		{
+			CreateClipData(playable);
+			base.OnBehaviourPlay(playable, info);
+		}
+
+		public override void ProcessFrame(Playable playable, FrameData info, object playerData)
+		{
+			CalculateCurrentClip(playable);
+
+			ClipData data = m_clipsData[m_currentClipIndex];
+			ScriptPlayable<DialogueBehaviour> playableInput = data.Input;
+			DialogueBehaviour input = playableInput.GetBehaviour();
+
+			ProcessCurrentClip(playable, playerData, input, data);
+		}
+
+		private void CalculateCurrentClip(Playable playable)
+		{
+			//TODO not very efficient, is there a way to find the current input?
+			double currentTime = playable.GetTime();
+			for (int i = 0; i < m_clipsData.Length; i++)
+			{
+				if (currentTime >= m_clipsData[i].Range.Start && currentTime < m_clipsData[i].Range.End)
+				{
+					m_currentClipIndex = i;
+					m_clipsData[m_currentClipIndex].state = ClipState.Pending;
+					break;
+				}
+			}
+		}
```

Comment:
Are you looking just to know which clip is currently played by the Timeline? You should use weights.

@rainaldisatria is doing it here for instance: https://github.com/UnityTechnologies/open-project-1/blob/c1aca3610d73fa83acb43e72e895f349c4250fbf/UOP1_Project/Assets/Scripts/Cutscenes/Dialogue%20Timeline/DialogueControlMix.cs

If the weight is 0, the clip is not being played. If it's 1, it's played and it's the only one.
More than one clip can be played at a time due to a blend. In which case the weight won't be 1 but something in between.